wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceInfoViewModel.swift

```diff
@@ -117,8 +118,11 @@ final class DeviceInfoViewModel: ObservableObject {
     @MainActor
     func enrollClient() async {
         self.isActionInProgress = true
-        let certificate = await actionsHandler.enrollClient()
-        self.e2eIdentityCertificate = certificate
+        if let certificate = await actionsHandler.enrollClient() {
+            self.e2eIdentityCertificate = certificate
+        } else {
+            showEnrollmentCertificateError = true
+        }
```

Comment:
If you let `enrollClient` throw we could we do 

```suggestion
        do {
            self.e2eIdentityCertificate = await actionsHandler.enrollClient()
        } catch {
            showEnrollmentCertificateError = true
        }
```

Reply:
done